ingest-app/src/cmr/ingest/services/ingest_service/variable.clj

```diff
@@ -1,15 +1,39 @@
 (ns cmr.ingest.services.ingest-service.variable
   (:require
+   [cheshire.core :as json]
+   [clojure.string :as string]
    [cmr.common.util :refer [defn-timed]]
    [cmr.transmit.metadata-db2 :as mdb2]
-   [cmr.umm-spec.umm-spec-core :as spec]))
+   [cmr.umm-spec.umm-spec-core :as spec]
+   [digest]))
+
+(defn- normalized-string
+  "Returns the given string with leading and trailing whitespaces trimmed
+  and converted to lowercase."
+  [value]
+  (some-> value
+          string/trim
+          string/lower-case))
+
+(defn- get-variable-fingerprint
+  "Returns the fingerprint of the given variable concept."
+  [variable-concept]
+  (let [parsed (json/decode (:metadata variable-concept) true)
+        {:keys [AcquisitionSourceName Name Units Dimensions]} parsed
+        id-string (format "%s|%s|%s|%s"
+                          (normalized-string AcquisitionSourceName)
+                          (normalized-string Name)
+                          (normalized-string Units)
+                          (pr-str (mapv #(update % :Name normalized-string) Dimensions)))]
```

Comment:
Are we 100% that we should consider Dimension order significant? I guess I will follow up with Jon Pals.

Reply:
I went with Jon's suggestion because: 1) Jon said that the order is significant. 2) Based on Chris L.'s comments, the variables will almost always be auto-generated which makes the change of the Dimension objects orders more unlikely. I have sent Jon an email for confirmation again, but he is off for the next couple of days. We might have to add more aliases to Dimension Names anyways based on Jon's comments. I expect that we have to change the fingerprint algorithm in the future even though that is something that we don't want to do.